chains/tendermint/query.go

```diff
@@ -145,50 +145,64 @@ func (c *Chain) QueryDenomTraces(ctx core.QueryContext, offset, limit uint64) (*
 }
 
-func (dst *Chain) QueryPacketAcknowledgement(ctx core.QueryContext, sequence uint64) ([]byte, error) {
-	txs, err := dst.QueryTxs(int64(ctx.Height().GetRevisionHeight()), 1, 1000, ackPacketQuery(dst.Path().ChannelID, int(sequence)))
+// queryReceivedPacket finds a RecvPacket event corresponding to `sequence` and return the packet in it
+func (c *Chain) queryReceivedPacket(ctx core.QueryContext, sequence uint64) (*chantypes.Packet, clienttypes.Height, error) {
+	txs, err := c.QueryTxs(int64(ctx.Height().GetRevisionHeight()), 1, 1000, recvPacketQuery(c.Path().ChannelID, int(sequence)))
 	switch {
 	case err != nil:
-		return nil, err
+		return nil, clienttypes.Height{}, err
 	case len(txs) == 0:
-		return nil, fmt.Errorf("no transactions returned with query")
+		return nil, clienttypes.Height{}, fmt.Errorf("no transactions returned with query")
 	case len(txs) > 1:
-		return nil, fmt.Errorf("more than one transaction returned with query")
+		return nil, clienttypes.Height{}, fmt.Errorf("more than one transaction returned with query")
+	}
+
+	packet, err := core.FindPacketFromEventsBySequence(txs[0].TxResult.Events, chantypes.EventTypeRecvPacket, sequence)
+	if err != nil {
+		return nil, clienttypes.Height{}, err
+	}
+	if packet == nil {
+		return nil, clienttypes.Height{}, fmt.Errorf("can't find the packet from events")
+	}
+
+	height := clienttypes.NewHeight(0, uint64(txs[0].Height))
+
+	return packet, height, nil
+
+}
+
+// queryWrittenAcknowledgement finds a WriteAcknowledgement event corresponding to `sequence` and returns the acknowledgement in it
+func (dst *Chain) queryWrittenAcknowledgement(ctx core.QueryContext, sequence uint64) ([]byte, clienttypes.Height, error) {
+	txs, err := dst.QueryTxs(int64(ctx.Height().GetRevisionHeight()), 1, 1000, writeAckQuery(dst.Path().ChannelID, int(sequence)))
+	switch {
+	case err != nil:
+		return nil, clienttypes.Height{}, err
+	case len(txs) == 0:
+		return nil, clienttypes.Height{}, fmt.Errorf("no transactions returned with query")
+	case len(txs) > 1:
+		return nil, clienttypes.Height{}, fmt.Errorf("more than one transaction returned with query")
 	}
 
 	ack, err := core.FindPacketAcknowledgementFromEventsBySequence(txs[0].TxResult.Events, sequence)
 	if err != nil {
-		return nil, err
+		return nil, clienttypes.Height{}, err
 	}
 	if ack == nil {
-		return nil, fmt.Errorf("can't find the packet from events")
+		return nil, clienttypes.Height{}, fmt.Errorf("can't find the packet from events")
 	}
-	return ack.Data(), nil
-}
 
-// QueryPacketReciept returns the packet reciept proof at a given height
-func (c *Chain) QueryPacketReciept(ctx core.QueryContext, seq uint64) (recRes *chantypes.QueryPacketReceiptResponse, err error) {
-	return chanutils.QueryPacketReceipt(c.CLIContext(int64(ctx.Height().GetRevisionHeight())), c.PathEnd.PortID, c.PathEnd.ChannelID, seq, true)
+	height := clienttypes.NewHeight(0, uint64(txs[0].Height))
```

Comment:
Can't we set a non-zero revision number here?

Reply:
Fixed: https://github.com/hyperledger-labs/yui-relayer/pull/90/commits/04d77bcdd90a1e6f6385fe46d33f126e86025b4a#diff-b9642f5116a057732bc245b582fcd98b7dcd3a6f0ceb83789ad6116f3da4b778R300